### PR TITLE
Hide not owned repos from sidebar and repo list

### LINKED ISSUE #971: Synchronize status between active view, the right pane, and the top (navigation) bar
### Component

web-ui

### Describe the bug

When a pipeline is triggered, if I browse over the view/screen where the steps are displayed, when the pipeline finishes, the interface "gives the impression" the pipeline is still in progress based on some indicators, specifically, there will still be a section with the `user/repository` (with the blue two-dots animation) and the circle in the top (navigation) bar.

Notice that when it finishes, a section with the `user/repository` is created as well right before the top one with the correct "finished" status.

So it's just a matter of synchronizing some elements and hiding/removing them from the DOM — my guess :thinking:

### System Info

```shell
Codeberg's Woodpecker CI
```

### Additional context

_No response_

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

### LINKED ISSUE #1454: do not send all public pipeline updates to all the users
Currently, all the public pipeline updates are pushed to every user and get filtered on the client-side. This creates lots of unnecessary traffic for servers with many (authenticated + unauthenticated) users and/or active public repositories.

See #1453 and [#1436](https://github.com/woodpecker-ci/woodpecker/pull/1436#issuecomment-1328218722)

### PR DESCRIPTION
**Pull request by anbraten**
closes #971 , closes #1454
closes #1436 

fixes: https://codeberg.org/Codeberg-CI/feedback/issues/28

## Notes for testing:
check if the user gets pipeline updates (not only the sidebar, but most importantly new pipelines and updates to the workflows and steps)

- for own private projects
- for org private projects
- for public projects the user does not directly own, but is allowed to see
- for public projects as unauthenticated users

### CODE REVIEW

File: web/src/compositions/useEvents.ts [hunk 1]
```diff
@@ -30,7 +29,6 @@ export default () => {
     }
     const { pipeline } = data;
     pipelineStore.setPipeline(repo.owner, repo.name, pipeline);
-    pipelineStore.setPipelineFeedItem({ ...pipeline, name: repo.name, owner: repo.owner, full_name: repoSlug(repo) });
```
Comment (anbraten) [inline web/src/compositions/useEvents.ts:33]: Not needed anymore as we store a single list of all pipelines.

File: web/src/views/repo/RepoWrapper.vue [hunk 2]
```diff
@@ -132,7 +132,7 @@ watch([repoOwner, repoName], () => {
   loadRepo();
 });
 
-const badgeUrl = computed(() => `/api/badges/${repo.value.owner}/${repo.value.name}/status.svg`);
+const badgeUrl = computed(() => `/api/badges/${repo.value?.owner}/${repo.value?.name}/status.svg`);
```
Comment (lukashass) [inline web/src/views/repo/RepoWrapper.vue:135]: When `repo` is undefined the `badgeUrl` will be invalid. I would suggest hiding it in that case.

File: web/src/store/repos.ts [hunk 3]
```diff
@@ -1,44 +1,54 @@
...
+    repos.set(repoSlug(repo), repo);
+    return repo;
+  }
+
+  async function loadRepos() {
+    const _repos = await apiClient.getRepoList();
+    _repos.forEach((repo) => {
+      repos.set(repoSlug(repo), repo);
+    });
+    ownedRepoSlugs.value = _repos.map((repo) => repoSlug(repo));
```
Comment (lukashass) [inline web/src/store/repos.ts:42]: @anbraten How is `ownedRepos` different from `repos` when both are derived from `apiClient.getRepoList()`?

File: web/src/store/pipelines.ts [hunk 4]
```diff
@@ -1,99 +1,103 @@
...
+            <PipelineFeed>{
+              ...pipeline,
+              full_name: _repoSlug,
+              owner: _repoSlug.split('/')[0],
+              name: _repoSlug.split('/')[1],
+              number: _pipelineNumber,
+            },
+        );
+        return [...acc, ...repoPipelinesArray];
+      }, [] as PipelineFeed[])
```
Comment (lukashass) [inline web/src/store/pipelines.ts:77]: @anbraten Can't you pass `PipelineFeed[]` as type parameter to `reduce`?

File: web/src/store/repos.ts (hunk 3 above)
Comment (anbraten) [reply web/src/store/repos.ts:42]: @lukashass Its the same in this moment. The difference is that we add repos (not owned ones) when we receive an event for it see useEvents.

File: web/src/store/pipelines.ts (hunk 4 above)
Comment (anbraten) [reply web/src/store/pipelines.ts:77]: changed it

File: web/src/views/repo/RepoWrapper.vue (hunk 2 above)
Comment (anbraten) [reply web/src/views/repo/RepoWrapper.vue:135]: done

File: web/src/store/repos.ts (hunk 3 above)
Comment (lukashass) [reply web/src/store/repos.ts:42]: @anbraten What is the behavior we want eventually? Display public repos, never show public repos or let the user toggle? I feel like rest and events should return the same thing :thinking:

Comment (anbraten) [reply web/src/store/repos.ts:42]: @lukashass We want both. The sidepanel and the repos page should show owned repos. In addition a user and anonymous users should be able to access public repos of other users / orgs. Those public repos are added on demand to the store if the corresponding pages are opened.

Comment (lukashass) [inline web/src/store/repos.ts:42]:
```suggestion
    const _ownedRepos = await apiClient.getRepoList();
    _ownedRepos.forEach((repo) => {
      repos.set(repoSlug(repo), repo);
    });
    ownedRepoSlugs.value = _ownedRepos.map((repo) => repoSlug(repo));
```

@anbraten What do you think of this?

**Comment by lukashass** [comment]
Tested:

- [x] for own private projects
- [x] for org private projects
- [x] for public projects the user does not directly own, but is allowed to see
- [x] for public projects as unauthenticated users


Findings:

- Org private projects do not appear in the feed
- Woodpecker admin sees all repos/pipeline updates

@anbraten

**Comment by anbraten** [comment]
> when starting on route /org/repo the feed is empty and doesn't receive updates. When starting on /repos the feed shows pipelines and receives updates.

Strange. It should be have a pipeline entry to add the step updates to, but I will investigate on that.

> Woodpecker admin sees all repos/pipeline updates

Are you sure that is the case? The filter I've added only shows pipelines which match the names from `GetRepos` and that function uses `RepoList` from the store which is not checking for admin permission as far as I can tell.

@lukashass

**Comment by anbraten** [comment]
@lukashass Kindly reminder

**Comment by lukashass** [comment]
> Are you sure that is the case? The filter I've added only shows pipelines which match the names from GetRepos and that function uses RepoList from the store which is not checking for admin permission as far as I can tell.

@anbraten Well, just tried this again and can't reproduce. I guess I messed something up before.

The other thing is still an issue though!

**Comment by anbraten** [comment]
@lukashass Thanks for testing. The problem was that the repo list was not loaded on some pages. Should be fixed now.

File: web/components.d.ts [hunk 5]
```diff
@@ -77,6 +77,7 @@ declare module '@vue/runtime-core' {
     IPhXCircle: typeof import('~icons/ph/x-circle')['default']
     ISimpleIconsGitea: typeof import('~icons/simple-icons/gitea')['default']
     ITeenyiconsGitSolid: typeof import('~icons/teenyicons/git-solid')['default']
+    ITeenyiconsRefreshOutline: typeof import('~icons/teenyicons/refresh-outline')['default']
```
Comment (lukashass) [inline web/components.d.ts:80]: @anbraten What's this change about?

Comment (anbraten) [reply web/components.d.ts:80]: @lukashass Its an automatically created file adding typings for all components. It has some issues detecting the icon components, so each time you open a page or so with an icon not being indexed before it adds that component. We just approve it in reviews. 🙈

**Review by lukashass — APPROVED** [review]

**Pull request merged**